src/rebar_prv_update.erl

```diff
@@ -34,7 +34,10 @@ init(State) ->
 -spec do(rebar_state:t()) -> {ok, rebar_state:t()} | {error, string()}.
 do(State) ->
     Names = rebar_packages:get_all_names(State),
-    [rebar_packages:update_package(Name, State) || Name <- Names],
+    Resources = rebar_state:resources(State),
+    #{repos := RepoConfigs,
+      base_config := BaseConfig} = rebar_resource:find_resource_state(pkg, Resources),
+    [[update_package(Name, maps:merge(RepoConfig, BaseConfig), State) || RepoConfig <- RepoConfigs] || Name <- Names],
```

Comment:
Since both generators are from external vars, I think you'll have the same result (but with fewer  intermediary lists created) if you call:

```
_ = [update_package(Name, maps:merge(RepoConfig, BaseConfig), State) || Name <- Names, RepoConfig <- RepoConfigs],
... 
```

Reply:
Ah yea, ok, had meant to clean this up.

Reply:
Does httpc have a pool of keep alive connections or does it only keep one and if you then use the same profile for hitting a different host it creates a new connection and drops the other?

Reply:
I think there's a param to set how many it keeps in the pool